File: plugins/devkit/devkit-java-tests/testData/inspections/statefulEp/FakeFile.java
import com.intellij.openapi.project.Project;
import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiReference;

public class FakeFile {
  final PsiElement pe;
  final PsiReference r;
  Project p;
  final Project pf;

  public FakeFile(Project project) {
    super();
    pe = null;
    r = null;
    p = pf = project;
  }
}

File: plugins/devkit/devkit-kotlin-tests/testData/inspections/statefulEp/FakeFile.kt
import com.intellij.openapi.project.Project
import com.intellij.psi.PsiElement
import com.intellij.psi.PsiReference

class FakeFile(val pf: Project) {
  val pe: PsiElement? = null
  val r: PsiReference? = null
  var p: Project

  init {
    p = pf
  }
}